internal/relationtuple/transact_server.go

```diff
@@ -170,15 +167,28 @@ func (h *handler) createRelation(w http.ResponseWriter, r *http.Request, _ httpr
 //       204: emptyResponse
 //       400: genericError
 //       500: genericError
-func (h *handler) deleteRelation(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-	rel, err := (&InternalRelationTuple{}).FromURLQuery(r.URL.Query())
+func (h *handler) deleteRelations(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	q := r.URL.Query()
+	query, err := (&RelationQuery{}).FromURLQuery(q)
 	if err != nil {
 		h.d.Writer().WriteError(w, r, herodot.ErrBadRequest.WithError(err.Error()))
 		return
 	}
 
-	if err := h.d.RelationTupleManager().DeleteRelationTuples(r.Context(), rel); err != nil {
-		h.d.Logger().WithError(err).WithFields(rel.ToLoggerFields()).Errorf("got an error while deleting the relation tuple")
+	l := h.d.Logger()
+	for k := range q {
+		l = l.WithField(k, q.Get(k))
+	}
+	l.Debug("deleting relation tuples")
+
+	rels, _, err := h.d.RelationTupleManager().GetRelationTuples(r.Context(), query)
```

Comment:
This is quite problematic as you only delete the first page here. Can you instead implement the selection of delete candidates on the persistence layer? A request DELETE `/relation-tuples?namespace=foo` should result in the following SQL statement to be executed: `DELETE FROM keto_relation_tuples WHERE namespace=foo`.

Reply:
Aha, yeah I will try. For the gRPC part, I may need time to understand it because I've never worked with gRPC before.

Reply:
Ah so there is no specific delete RPC, but only the transact one. I would add a `DeleteReleationTuples` RPC here https://github.com/ory/keto/blob/67c4c608a9bbdb866701e7f3746c1d13f2e6f2f0/proto/ory/keto/acl/v1alpha1/write_service.proto#L14-L20
It should behave exactly as the REST handler, so you can probably use the same internal helper function. The current transaction handler is implemented here: https://github.com/ory/keto/blob/67c4c608a9bbdb866701e7f3746c1d13f2e6f2f0/internal/relationtuple/transact_server.go#L30-L53, you can just follow the implementation of that as a guidance.